graders/python/python_autograder/code_feedback.py

```diff
@@ -304,6 +304,110 @@ def bad(msg):
 
         return True
 
+    @classmethod
+    def check_dict(
+        cls,
+        name,
+        ref,
+        data,
+        partial_keys=None,
+        check_keys=False,
+        check_values=False,
+        key_type=None,
```

Comment:
`key_type` and `value_type` don't feel that useful to me. Why not just have the types related to the `ref` type? E.g. `type(ref["foo"]) == type(data["foo"])`

Reply:
I would prefer removing them and replacing them with a different name; e.g. `typecheck = "keys" | "values" | "all" | "none"`, which matches against ref.

Reply:
I think using an enum makes things a lot cleaner here as well.

Reply:
I'm thinking that all values will be typechecked, and that this doesn't need to be configured.

Reply:
I'm also not even sure how key type checking is useful:
```
>>> {1:2}.keys()
dict_keys([1])
>>> {'1':2}.keys()
dict_keys(['1'])
```